src/syscheckd/create_db.c

```diff
@@ -814,6 +827,7 @@ void fim_whodata_event(whodata_evt * w_evt) {
 
     struct stat file_stat;
 
+    w_rwlock_rdlock(&syscheck.directories_lock);
```

Comment:
Could we narrow the scope of this lock a little? Place it around `fim_checker` and `fim_process_missing_entry`